### PR TITLE
Add testing for foreach scalar Tensor overloads in inductor

### PR DESCRIPTION
**Pull request by janeyx99**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111600

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by mlazos — APPROVED** [review]

File: test/inductor/test_foreach.py [hunk 1]
```diff
@@ -132,6 +145,12 @@ def test_single_list(self, op):
     def test_single_scalar(self, op):
         self._test_single_scalar(op)
         self.assertEqual(torch._inductor.metrics.generated_kernel_count, 1)
+
```
Comment (janeyx99) [inline test/inductor/test_foreach.py:148]:
```suggestion

```

File: test/inductor/test_foreach.py [hunk 2]
```diff
@@ -115,6 +116,18 @@ def fn(a0, a1):
                 torch.rand(20, 20, device="cuda:0"),
             ),
         )
+
```
Comment (janeyx99) [inline test/inductor/test_foreach.py:119]:
```suggestion

```

File: test/inductor/test_foreach.py [hunk 3]
```diff
@@ -56,6 +56,7 @@
 )
 bin_ops = parametrize("op", bin_ops_under_test, name_fn=lambda f: f.__name__)
 scalar_bin_ops = parametrize("op", bin_ops_under_test[:4], name_fn=lambda f: f.__name__)
+scalar_tensor_bin_ops = parametrize("op", bin_ops_under_test[:2], name_fn=lambda f: f.__name__)
```
Comment (janeyx99) [inline test/inductor/test_foreach.py:59]:
```suggestion
scalar_tensor_bin_ops = parametrize(
    "op", bin_ops_under_test[:2], name_fn=lambda f: f.__name__
)
```

File: test/inductor/test_foreach.py [hunk 4]
```diff
@@ -132,6 +146,11 @@ def test_single_list(self, op):
     def test_single_scalar(self, op):
         self._test_single_scalar(op)
         self.assertEqual(torch._inductor.metrics.generated_kernel_count, 1)
+    @requires_cuda()
```
Comment (janeyx99) [inline test/inductor/test_foreach.py:149]:
```suggestion

    @requires_cuda()
```

File: test/inductor/test_foreach.py [hunk 5]
```diff
@@ -115,6 +118,17 @@ def fn(a0, a1):
                 torch.rand(20, 20, device="cuda:0"),
             ),
         )
+    def _test_single_scalar_tensor(self, op):
```
Comment (janeyx99) [inline test/inductor/test_foreach.py:121]:
```suggestion

    def _test_single_scalar_tensor(self, op):
```

**Comment by janeyx99** [comment]
:( sad

if this change cannot wait 3.5 weeks, feel free to land this for me as i won’t have access to a computer

**Pull request closed without merging**